libgnucash/engine/gnc-commodity.cpp

```diff
@@ -1592,7 +1593,7 @@ gnc_commodity_table *
 gnc_commodity_table_new(void)
 {
     gnc_commodity_table * retval = g_new0(gnc_commodity_table, 1);
-    retval->ns_table = g_hash_table_new(&g_str_hash, &g_str_equal);
+    StrCommodityNSMap().swap (retval->ns_table);
```

Comment:
As for StrCommodityMap, better to use placement new.

Reply:
> As for StrCommodityMap, better to use placement new.

or `auto retval = new gnc_commodity_table;` which will properly initialize the map.

Reply:
It will, but you'll need an explicit `~gnc_commodity_table()` to clean up the GList. Note that mixing `new` with `malloc` like that is considered risky. The compiler will probably do what you want but it doesn't have to.